lib/bindConnectors/bindMessage.js

```diff
@@ -1,15 +1,98 @@
-var when             = require('when');
-var _ 				 = require('lodash');
+var when              = require('when');
+var _ 				  = require('lodash');
+const { performance } = require('perf_hooks');
+const { client, v2 }  = require('@datadog/datadog-api-client');
+
 
 const { camelCase }  = require('../utils/mout');
 var getMissingParams = require('./getMissingParams');
 var formatInputBody  = require('./formatInputBody');
 const interceptAfterSuccess = require('./interceptAfterSuccess');
+const { initialiseLogger } = require('../logger');
+const { getDatadogSecrets } = require('./getConfigs');
 
 
-module.exports = function (message, threadneedle, connectorConfig) {
+const initDataDogApi = async () => {
```

Comment:
[personal nit] I've never understood why in JS, devs define an arrow/anonymous function, only to then assign it to a const?
If we're doing named functions, then use `function` for that - it's what it's there for 🙃

Reply:
When you define a function using the function keyword it is hoisted to the top of the scope, this doesn't happen when you assign the function to a const variable which is block scoped

Reply:
You're right - but is that an issue?

Functions are rarely defined within block scope like if/while stmt, but even if they were, given we have a functional style of programming (or at least partially), that shouldn't be an issue. And the scope in most cases are the file/module itself, or the function scope the sub-function is defined within. 
AFAICT, I don't think the scope, and therefore the hoisting, poses a problem in most of our code bases.

Reply:
I don't think in this scenario it is an issue, but that doesn't matter. What does matter is 1. consistency, so I always write my functions in this way and 2. not making assumptions, I don't assume that the codebase is safe for me to hoist my functions I just block scope them because I know this will not create an issue.

The benefit of this is that I no longer have to care about this as an issue it occupies no space in my mind. The way you say that we "partially" use functional programming and that it wouldn't be a problem in "most" of our code bases tells me that you don't have the same level of confidence.